spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/pulsar/PulsarPropertiesTests.java

```diff
@@ -390,8 +400,34 @@ class TemplateProperties {
 		void bind() {
 			Map<String, String> map = new HashMap<>();
 			map.put("spring.pulsar.template.observations-enabled", "true");
+			map.put("spring.pulsar.template.transaction.enabled", "true");
+			map.put("spring.pulsar.template.transaction.required", "true");
+			map.put("spring.pulsar.template.transaction.timeout", "60s");
 			PulsarProperties.Template properties = bindPropeties(map).getTemplate();
 			assertThat(properties.isObservationsEnabled()).isTrue();
+			assertThat(properties.getTransaction().isEnabled()).isTrue();
+			assertThat(properties.getTransaction().isRequired()).isTrue();
+			assertThat(properties.getTransaction().getTimeout()).isEqualTo(Duration.ofSeconds(60));
+		}
+
+	}
+
+	@Nested
+	class TransactionProperties {
+
+		@ParameterizedTest
+		@MethodSource
+		void transactionsEnabledTests(boolean listenerEnablesTransactions, boolean templateEnablesTransactions,
```

Comment:
I am on the fence here w/ the use of `@ParameterizedTest` w/ 4 items in the test arguments stream provider vs. having 4 individual (yet repetitive) tests methods (feels a bit ""six of one, half a dozen of the other"). 

On one hand, the 4 individual test methods makes it clear what is being tested. And to get that same clarity w/ this impl we have to add the test name as arg0 and then do `@ParameterizedTest(name = "{0}")`. 
On the other hand, this current impl reduces the 4 tests that are about 99% the same. 🤷🏻

Reply:
Yeh, I think the value for doing this for 4 cases does not outweigh having 4 concise test methods (delegating to a single test impl method). I will adjust.

Reply:
Done